server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -227,7 +227,7 @@ protected void masterOperation(
 
         IndicesStatsRequest statsRequest = new IndicesStatsRequest().indices(rolloverRequest.getRolloverTarget())
             .clear()
-            .indicesOptions(IndicesOptions.fromOptions(true, false, true, true))
+            .indicesOptions(buildIndicesOptions(true, true, rolloverRequest.indicesOptions().failureStoreOptions()))
```

Comment:
Hmm I thought I double-checked functionality being the same, I guess I missed this... And yeah I knew that was what you were thinking of (and realized that that didn't work exactly), so thought I'd give this a try but I agree that this is not ideal either. Will revert 👍🏻

Reply:
On second thought, I now remember why I exacted this `buildIndicesOptions` method: my goal was to add the `FailureStoreOptions` to the `IndicesOptions` for the stats request, but I wanted to avoid having two of those bulky `IndicesOptions` constructions. I'm thinking of either adding an `allowEmptyExpressions` parameter to the method (because you rightfully noticed that I made a mistake while extracting), or I keep the two `IndicesOptions` constructions separate (and maybe extract the construction of the options for the stats request to a method because the `masterOperation` method is already rather long). Any thoughts?

Reply:
Hmm, while the refactoring lets us add new properties to indices options easier than before, it does make the construction a bit bulky. Maybe we could check the defaults against what we're configuring and if they match, let's not specify the property? Does that buy us a reduction in size at all?